nvfuser/__init__.py

```diff
@@ -177,13 +185,21 @@ def execute(
             self.fake_inputs = [fake_mode.from_tensor(inp) for inp in inputs]
 
         results = None
+
         try:
+            if len(_enable_options) or len(_disable_options):
+                warnings.warn(
+                    "Reset the FusionCache manually to avoid reusing kernels when re-executing the fusion definition with different options."
+                )
+
             results = self._execute(
                 inputs,
                 device=device,
                 override_user_schedule=override_user_schedule,
                 capture_debug_output=capture_debug_output,
                 profile=profile,
+                _enable_options=_enable_options,
+                _disable_options=_disable_options,
```

Comment:
As discussed offline, moving forward with renaming the kwargs to `_enable/_disable_options`. I added a warning about the possibility of kernel reuse so that we are aware of this when utilizing this feature.